Source/Core/DolphinQt2/Config/CheatWarningWidget.cpp

```diff
@@ -0,0 +1,83 @@
+// Copyright 2017 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#include "DolphinQt2/Config/CheatWarningWidget.h"
+
+#include <QHBoxLayout>
+#include <QLabel>
+#include <QMessageBox>
+#include <QPixmap>
+#include <QPushButton>
+#include <QStyle>
+#include <QTimer>
+
+#include "Core/ConfigManager.h"
+#include "Core/Core.h"
+#include "DolphinQt2/Settings.h"
+
+CheatWarningWidget::CheatWarningWidget(const std::string& game_id) : m_game_id(game_id)
+{
+  CreateWidgets();
+  ConnectWidgets();
+
+  setHidden(true);
+
+  connect(&Settings::Instance(), &Settings::EnableCheatsChanged, this, &CheatWarningWidget::Update);
+  connect(this, &CheatWarningWidget::EmulationStarted, [this] { Update(true); });
+  connect(this, &CheatWarningWidget::EmulationStopped, [this] { Update(false); });
+
+  Update(Core::IsRunning());
+}
+
+void CheatWarningWidget::CreateWidgets()
+{
+  auto* icon = new QLabel;
+
+  QPixmap warning_icon = style()->standardIcon(QStyle::SP_MessageBoxWarning).pixmap(24, 24);
+
+  icon->setPixmap(warning_icon);
+
+  m_text = new QLabel();
+  m_config_button = new QPushButton(tr("Configure Dolphin"));
+
+  m_config_button->setHidden(true);
+
+  auto* layout = new QHBoxLayout;
+
+  layout->addWidget(icon);
+  layout->addWidget(m_text, 1);
+  layout->addWidget(m_config_button);
+
+  layout->setContentsMargins(0, 0, 0, 0);
+
+  setLayout(layout);
+}
+
+void CheatWarningWidget::Update(bool running)
+{
+  bool hide_widget = true;
+  bool hide_config_button = true;
+
+  if (running && SConfig::GetInstance().GetGameID() == m_game_id)
+  {
+    hide_widget = false;
+    m_text->setText(tr("Changing cheats will only take effect when the game is restarted."));
+  }
+
+  if (!SConfig::GetInstance().bEnableCheats)
```

Comment:
This should read `EnableCheats` from `Settings`, not from `SConfig`, since it's already using it to connect to the signal.